recipes/_configure_server.rb

```diff
@@ -246,16 +246,19 @@
   auto_generated_pool_size = CernerSplunk.convert_to_bytes license_pools['auto_generated_pool_size']
   server_stanzas['lmpool:auto_generated_pool_enterprise']['quota'] = auto_generated_pool_size
   allotted_pool_size = 0
-
-  license_pools['pools'].each do |pool, pool_config|
-    pool_max_size = CernerSplunk.convert_to_bytes pool_config['size']
-    server_stanzas["lmpool:#{pool}"] = {
-      'description' => pool,
-      'quota' => pool_max_size,
-      'slaves' => pool_config['GUIDs'].join(','),
-      'stack_id' => 'enterprise'
-    }
-    allotted_pool_size += pool_max_size
+  ruby_block 'get_license_type' do
+    block do
+      license_pools['pools'].each do |pool, pool_config|
+        pool_max_size = CernerSplunk.convert_to_bytes pool_config['size']
+        server_stanzas["lmpool:#{pool}"] = {
+          'description' => pool,
+          'quota' => pool_max_size,
+          'slaves' => pool_config['GUIDs'].join(','),
+          'stack_id' => node.run_state['type']
+        }
+        allotted_pool_size += pool_max_size
```

Comment:
Since this moves the allotted_pool_size summation from compile time to execution time, I believe that makes the run state for the total_allotted_pool_size incorrect (since the node.run_state is set below, outside the ruby_block at compile time).  Then the error checking on line 48-51 of the license_server recipe is no longer accurate.

If one part of it moves to execution time, then all the parts have to move.  But then it won't fail the chef run until part way through, which makes that error checking considerably less useful since the bad server conf would have already been written out.  Will Splunk fail to start if the pools configured in server.conf are greater than the available licenses?

Reply:
I checked and Splunk doesn't seem to care if the total allocation of the pools is greater than the amount of the license stack. Although if you create or edit a pool via the UI and try to make it larger, something in the web console will set the pool size equal to the amount of available license when it saves it.

Reply:
I think the easiest fix here is to remove the ruby block here and simply re-order the operations in the license_server recipe so the data bag is processed prior to including this recipe. Just move line 18 down between lines 46 and 48 in the license_server recipe.